ci/check-documentation.py

```diff
@@ -39,12 +39,44 @@ def find_md_files():
 def main():
     all_passed = True
 
+    index_file_paths = (
+        OMZ_ROOT / 'models/intel/index.md',
+        OMZ_ROOT / 'models/public/index.md',
+        OMZ_ROOT / 'demos/README.md',
+    )
+
+    all_md_files = tuple(find_md_files())
+
     def complain(message):
         nonlocal all_passed
         all_passed = False
         print(message, file=sys.stderr)
 
-    for md_path in sorted(find_md_files()):
+    index_child_md_links = {}
+    for index_file_path in index_file_paths:
+        if not index_file_path.exists():
+            complain(f'{index_file_path}: file not found')
+            continue
+
+        required_md_links = []
+        for md_file in all_md_files:
+            if md_file.name == "README.md" and md_file.parent != index_file_path.parent:
+                try:
+                    md_rel_path = md_file.relative_to(index_file_path.parent)
+                except ValueError:
+                    continue
+
+                md_intermediate_parents = list(md_rel_path.parents)[1:-1] # removed root and first parent dirs
+
+                if not any((index_file_path.parent / parent_dir / 'README.md').exists()
+                    for parent_dir in md_intermediate_parents):
```

Comment:
This should be indented by one more level, because otherwise it looks like it's part of the body of the `if` statement.

Reply:
Done.